File: python/QuotaCalc.py
#!/usr/bin/env python
# coding: utf-8

import math
def check_user_input(input):
    try:
        # Convert it into integer
        val = int(input)
        print("Input is an integer number. Number = ", val)
        return(val)
    except ValueError:
        print("No.. input is not a number. It's a string")

V = input("Enter in the valid number of votes ")
v = check_user_input(V)
S = input("Enter in the number of seats to allocate ")
s = check_user_input(S)
Q =(v/s+1)
print("the Quota is ",Q) 

File: c/quotaCalc.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>

/* Calculate the quota for a PR-STV Election
 The Droop Quota is
 Q = (n/r+1) + 1
 The Hare Quota is
 Q = (n/r) 
*/

/* DroopQuota Calculate */
int droopQuota ( n, r){
int quota;
quota = round(((n/(r+1))+1));

return quota;
}

/* Hare Quota Calculate */
int hareQuota ( n, r){
int quota;
quota = round(n/r);

return quota;
}

/* Main Function */
int main(){
int n, r;

    /* get number of votes */
    printf("Enter a number votes : ");
    scanf("%d",&n);

    printf("Enter a number seats to allocate : ");
    scanf("%d",&r);
    
    printf(" The Droop Quota would be %d\n", droopQuota(n,r));
    printf(" The Hare Quota would be %d\n", hareQuota(n,r));

}
